cpp/src/slice2cpp/Gen.cpp

```diff
@@ -2113,84 +2113,106 @@ Slice::Gen::DataDefVisitor::visitExceptionStart(const ExceptionPtr& p)
     H << nl << "public:";
     H.inc();
 
-    H << nl << "using " << baseClass << "::" << baseName << ";";
-
     if (!allDataMembers.empty())
     {
-        H << sp;
-        H << nl << "/**";
-        H << nl << " * One-shot constructor to initialize all data members.";
-        for (const auto& dataMember : allDataMembers)
+        if (base && dataMembers.empty())
         {
-            map<string, CommentPtr>::iterator r = allComments.find(dataMember->name());
-            if (r != allComments.end())
-            {
-                H << nl << " * @param " << fixKwd(r->first) << " " << getDocSentence(r->second->overview());
-            }
+            // Reuse the base class constructors.
+            H << nl << "using " << baseClass << "::" << baseName << ";";
         }
-        H << nl << " */";
-        H << nl << name << "(";
-
-        for (vector<string>::const_iterator q = allParamDecls.begin(); q != allParamDecls.end(); ++q)
+        else
         {
-            if (q != allParamDecls.begin())
+            H << nl << "/**";
+            H << nl << " * Default constructor.";
+            H << nl << " */";
+            H << nl << name << "() noexcept = default;";
+
+            H << sp;
+            H << nl << "/**";
+            H << nl << " * One-shot constructor to initialize all data members.";
```

Comment:
I didn't change the code for this one-shot ctor. git is getting confused by the change in indentation.